test/health_cards/qbp_client_test.rb

```diff
@@ -0,0 +1,166 @@
+
+require 'test_helper'
+require 'health_cards/qbp_client'
+
+class QBPClientTest < ActiveSupport::TestCase
+  setup do
+    @complete_patient = { patient_list: { id: 'J19X5',
+                                          assigning_authority: 'AIRA-TEST',
+                                          identifier_type_code: 'MR' },
+                          patient_name: { family_name: 'WeilAIRA',
+                                          given_name: 'BethesdaAIRA',
+                                          second_or_further_names: 'Delvene',
+                                          suffix: '' },
+                          mothers_maiden_name: { family_name: 'WeilAIRA',
+                                                 given_name: 'BethesdaAIRA',
+                                                 name_type_code: 'M' },
+                          patient_dob: '20170610',
+                          admin_sex: 'F',
+                          address: { street: '1113 Wollands Kroon Ave',
+                                     city: 'Hamburg',
+                                     state: 'MI',
+                                     zip: '48139',
+                                     address_type: 'P' },
+                          phone: { area_code: '810',
+                                   local_number: '2499010' } }
+
+    WebMock.allow_net_connect!
+  end
+
+  def teardown
+    WebMock.disable_net_connect!
+  end
+
+  # General Functionality Tests
+
+  test 'query() method successfully returns an HL7 Message' do
+    v2_response_body = HealthCards::QBPClient.query({})
+    assert_instance_of(HL7::Message, v2_response_body)
+  end
+
+  test 'translate() method successfully returns a stringified JSON object' do
+    v2_response_body = HealthCards::QBPClient.query({})
+    fhir_response_body = HealthCards::QBPClient.translate(v2_response_body)
+    parsed_fhir_response = begin
+      JSON.parse(fhir_response_body)
+    rescue StandardError
+      nil
+    end
+    assert_not_nil(parsed_fhir_response)
+  end
+
+  test 'raises error if inputted sandbox credentials are incorrectly formatted' do
+    user_sandbox_credentials = { username: 'test_user', password: 'test_password', facilityID: 'test_facilityID' }
+    missing_credential = user_sandbox_credentials.except(:password)
+    assert_raises HealthCards::InvalidSandboxCredentialsError do
+      HealthCards::QBPClient.query({}, missing_credential)
+    end
+
+    non_string_credential = user_sandbox_credentials
+    non_string_credential[:password] = 1
+    assert_raises HealthCards::InvalidSandboxCredentialsError do
+      HealthCards::QBPClient.query({}, non_string_credential)
+    end
+  end
+
+  # Client Connectivity Tests
+
+  test 'Connectivity Test Works (Successfully connected to IIS Sandbox endpoint)' do
+    service_def = 'lib/assets/service.wsdl'
+    client = Savon.client(wsdl: service_def,
+                          endpoint: 'http://localhost:8081/iis-sandbox/soap',
```

Comment:
Should this be `endpoint: Rails.application.config.iisSandboxHost + '/iis-sandbox/soap'`?

Reply:
Unless you mock the external services, in which case hardcode the response I guess

Reply:
Good catch - I forgot to remove the hardcoded instance from the test file. Fixed with latest commit.